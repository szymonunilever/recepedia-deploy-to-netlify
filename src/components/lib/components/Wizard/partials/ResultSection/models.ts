export interface ResultSectionProps {
  title: string;
  subheading: string;
  [key: string]: string | number | boolean | object | null;
}
