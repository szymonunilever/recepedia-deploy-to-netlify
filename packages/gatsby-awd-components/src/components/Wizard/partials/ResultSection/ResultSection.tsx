import React, { Fragment, FunctionComponent } from 'react';
import { Button } from '../../../Button';
import { ResultSectionProps } from './models';
import { TagName, Text } from '../../../Text';
import theme from './ResultSection.module.scss';
import cx from 'classnames';

const ResultSection: FunctionComponent<ResultSectionProps> = ({
  content: { onResult, noResult },
  resultSize,
  children,
  actionCallback,
  isLoading = false,
  callbacks,
}) => {
  const { title, subheading, ctas } = resultSize > 0 ? onResult : noResult;

  const buttons = ctas.map(cta => {
    const onClick = () => {
      if (cta.type === 'next') {
        return actionCallback;
      } else {
        return callbacks[cta.type] ? callbacks[cta.type] : undefined;
      }
    };

    return (
      <Button
        key={`button-${cta.type}`}
        onClick={onClick()}
        className={`wizard__button wizard__button--primary`}
        content={cta.content}
      />
    );
  });
  return (
    <Fragment>
      {!isLoading && (
        <div className={cx(theme.wizard__title, 'wizard__title')}>
          <Text className="" tag={TagName.h2} text={title} />
          {subheading && (
            <Text className="subheading" tag={TagName.p} text={subheading} />
          )}
        </div>
      )}
      {children}
      {!isLoading && (
        <div key="wizardButtons" className="wizard__buttons">
          {buttons}
        </div>
      )}
    </Fragment>
  );
};

export default ResultSection;
