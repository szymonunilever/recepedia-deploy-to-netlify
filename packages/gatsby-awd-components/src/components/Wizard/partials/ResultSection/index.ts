export { default } from './ResultSection';
export { default as ResultSection } from './ResultSection';
export * from './models';
