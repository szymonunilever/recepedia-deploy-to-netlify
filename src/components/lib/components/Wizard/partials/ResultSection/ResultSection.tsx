import React, { Fragment, FunctionComponent } from 'react';
import { ResultSectionProps } from './models';
import { TagName, Text } from '../../../Text';

const ResultSection: FunctionComponent<ResultSectionProps> = ({
  title,
  subheading,
  children,
}) => {
  return (
    <Fragment>
      <div className="wizard__title">
        <Text className="" tag={TagName.h2} text={title} />
        {subheading && (
          <Text className="subheading" tag={TagName.p} text={subheading} />
        )}
      </div>
      {children}
    </Fragment>
  );
};

export default ResultSection;
