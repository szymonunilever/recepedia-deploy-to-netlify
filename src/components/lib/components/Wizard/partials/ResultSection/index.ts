export { default } from './ResultSection';
export { default as ResultSection } from './ResultSection';
