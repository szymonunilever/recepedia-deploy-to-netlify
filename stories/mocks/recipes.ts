/* eslint-disable */
// @ts-ignore
const recipe: Internal.Recipe = {
  id: 'c6ff2a45-402f-5542-9225-a0fab6041924',
  title: 'Bisteca acebolada',
  description: 'Bisteca acebolada',
  creationTime: new Date('2013-01-30T20:59:50Z'),
  recipeDetails: {
    serves: 4,
    totalTime: 30,
    preperationTime: 30,
    cookTime: 30,
  },
  ingredients: [],
  tagGroups: [
    {
      name: 'budgets',
      label: 'budgets',
      tags: [],
    },
    {
      name: 'cookingMethod',
      label: 'cookingMethod',
      tags: [],
    },
    {
      name: 'courses',
      label: 'courses',
      tags: [
        {
          id: 1400,
          name: 'prato principal',
        },
      ],
    },
    {
      name: 'cuisines',
      label: 'cuisines',
      tags: [
        {
          id: 1408,
          name: 'brasileira',
        },
      ],
    },
    {
      name: 'difficulties',
      label: 'difficulties',
      tags: [
        {
          id: 1460,
          name: 'dificuldade média',
        },
      ],
    },
    {
      name: 'dishes',
      label: 'dishes',
      tags: [
        {
          id: 1479,
          name: 'carne',
        },
      ],
    },
    {
      name: 'mainIngredient',
      label: 'mainIngredient',
      tags: [],
    },
    {
      name: 'occasions',
      label: 'occasions',
      tags: [
        {
          id: 1527,
          name: 'dia a dia',
        },
      ],
    },
    {
      name: 'seasons',
      label: 'seasons',
      tags: [],
    },
    {
      name: 'spiciness',
      label: 'spiciness',
      tags: [],
    },
    {
      name: 'timesOfDay',
      label: 'timesOfDay',
      tags: [
        {
          id: 617,
          name: 'almoço',
        },
        {
          id: 1552,
          name: 'jantar',
        },
      ],
    },
  ],
  localImage: {
    childImageSharp: {
      fluid: {
        base64:
          'data:image/jpeg;base64,/9j/2wBDAAYEBQYFBAYGBQYHBwYIChAKCgkJChQODwwQFxQYGBcUFhYaHSUfGhsjHBYWICwgIyYnKSopGR8tMC0oMCUoKSj/2wBDAQcHBwoIChMKChMoGhYaKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCj/wAARCAAKABQDASIAAhEBAxEB/8QAFwABAQEBAAAAAAAAAAAAAAAABgAEB//EACQQAAICAgIBAwUAAAAAAAAAAAECAwQFEQAxIQYScQcTIiVh/8QAFgEBAQEAAAAAAAAAAAAAAAAABQEC/8QAHhEAAgICAgMAAAAAAAAAAAAAAQIAAxFBBBIhUfD/2gAMAwEAAhEDEQA/AFeS/XehYMnWrtkL7MqyyTqCIdMNgp5VSRsdcy4JaWay1Onew9K5XAEsqtVQvHI2ipUjyvXxof3ho37dP1RjBUtWIBNI6yiKQr7xvptHyPnjT6kXbVTHVBVszwCX8ZPtyFfeNdHXfDEqdytitjGou94JKOuSd/eokyypXuvFBFDKo0SWmAPnzrXwRy5zLOEjOXlB0qtGoA6AESAAcubawAnxKvEJUHtP/9k=',
        aspectRatio: 1.95,
        width: 600,
        height: 307.6923076923077,
        src:
          'https://d37k6lxrz24y4c.cloudfront.net/e9dc924f238fa6cc29465942875fe8f0/572d686a-20a6-40d7-9f9e-da1e08c0421e-600.jpg',
        srcWebp:
          'https://d37k6lxrz24y4c.cloudfront.net/e9dc924f238fa6cc29465942875fe8f0/572d686a-20a6-40d7-9f9e-da1e08c0421e-600.webp',
        srcSet:
          'https://d37k6lxrz24y4c.cloudfront.net/e9dc924f238fa6cc29465942875fe8f0/572d686a-20a6-40d7-9f9e-da1e08c0421e-200.jpg 200w, https://d37k6lxrz24y4c.cloudfront.net/e9dc924f238fa6cc29465942875fe8f0/572d686a-20a6-40d7-9f9e-da1e08c0421e-300.jpg 300w, https://d37k6lxrz24y4c.cloudfront.net/e9dc924f238fa6cc29465942875fe8f0/572d686a-20a6-40d7-9f9e-da1e08c0421e-500.jpg 500w, https://d37k6lxrz24y4c.cloudfront.net/e9dc924f238fa6cc29465942875fe8f0/572d686a-20a6-40d7-9f9e-da1e08c0421e-600.jpg 600w, https://d37k6lxrz24y4c.cloudfront.net/e9dc924f238fa6cc29465942875fe8f0/572d686a-20a6-40d7-9f9e-da1e08c0421e-800.jpg 800w, https://d37k6lxrz24y4c.cloudfront.net/e9dc924f238fa6cc29465942875fe8f0/572d686a-20a6-40d7-9f9e-da1e08c0421e-900.jpg 900w',
        srcSetWebp:
          'https://d37k6lxrz24y4c.cloudfront.net/e9dc924f238fa6cc29465942875fe8f0/572d686a-20a6-40d7-9f9e-da1e08c0421e-200.webp 200w, https://d37k6lxrz24y4c.cloudfront.net/e9dc924f238fa6cc29465942875fe8f0/572d686a-20a6-40d7-9f9e-da1e08c0421e-300.webp 300w, https://d37k6lxrz24y4c.cloudfront.net/e9dc924f238fa6cc29465942875fe8f0/572d686a-20a6-40d7-9f9e-da1e08c0421e-500.webp 500w, https://d37k6lxrz24y4c.cloudfront.net/e9dc924f238fa6cc29465942875fe8f0/572d686a-20a6-40d7-9f9e-da1e08c0421e-600.webp 600w, https://d37k6lxrz24y4c.cloudfront.net/e9dc924f238fa6cc29465942875fe8f0/572d686a-20a6-40d7-9f9e-da1e08c0421e-800.webp 800w, https://d37k6lxrz24y4c.cloudfront.net/e9dc924f238fa6cc29465942875fe8f0/572d686a-20a6-40d7-9f9e-da1e08c0421e-900.webp 900w',
        sizes: '(max-width: 500px) 90vw, 500px',
        url:
          'https://scm-assets.constant.co/unilever/e9dc924f238fa6cc29465942875fe8f0/572d686a-20a6-40d7-9f9e-da1e08c0421e.jpg',
        title: 'Bisteca acebolada',
      },
    },
  },
  recipeId: 54311,
  fields: {
    slug: '/recipes/bisteca-acebolada',
  },
};

const recipes: Internal.Recipe[] = [
  { id: '1', ...recipe },
  { id: '2', ...recipe },
  { id: '3', ...recipe },
  { id: '4', ...recipe },
  { id: '5', ...recipe },
  { id: '6', ...recipe },
  { id: '7', ...recipe },
];

export default recipes;
