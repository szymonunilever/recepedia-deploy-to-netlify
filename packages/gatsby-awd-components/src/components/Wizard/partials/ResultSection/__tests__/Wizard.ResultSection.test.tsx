import { mount, ReactWrapper } from 'enzyme';
import React from 'react';
import ResultSection, { ResultSectionProps } from '../index';
import toJson from 'enzyme-to-json';
import resultContent from 'src/mocks/WizardResultSection.json';
import recipes from 'src/mocks/recipes.json';

describe('<ResultSection />', () => {
  let wrapper: ReactWrapper;
  const recipesList: Internal.Recipe[] = recipes.data.allRecipe.edges.map(
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    (item: { node: Internal.Recipe | any }) => item.node
  );

  const resultSectionProps: ResultSectionProps = {
    containerClass: 'wizard--result',
    stepId: 'result',
    resultSize: recipesList.length,
    ...resultContent,
    actionCallback: jest.fn(),
    callbacks: { test: jest.fn(() => {}) },
  };

  beforeEach(() => {
    wrapper = mount(
      <div className="wizard__content">
        <ResultSection {...resultSectionProps}>
          <>test</>
        </ResultSection>
      </div>
    );
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('matches the snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('matches the snapshot with 0 results', () => {
    expect(
      toJson(mount(<ResultSection {...resultSectionProps} resultSize={0} />))
    ).toMatchSnapshot();
  });
});
