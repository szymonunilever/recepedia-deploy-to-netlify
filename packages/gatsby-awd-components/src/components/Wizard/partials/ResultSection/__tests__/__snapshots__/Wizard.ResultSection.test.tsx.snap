// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResultSection /> matches the snapshot 1`] = `
<div
  className="wizard__content"
>
  <ResultSection
    actionCallback={[MockFunction]}
    callbacks={
      Object {
        "test": [MockFunction],
      }
    }
    containerClass="wizard--result"
    content={
      Object {
        "noResult": Object {
          "ctas": Array [
            Object {
              "content": Object {
                "label": "Tentar novamente",
              },
              "type": "back",
            },
          ],
          "subheading": "Minha chance de encontrar aumenta quando você não usa muitos filtros de uma vez.
Gostaria de tentar novamente?",
          "title": "=/ Busquei por toda cozinha mas não encontrei o que você procurava!",
        },
        "onResult": Object {
          "ctas": Array [
            Object {
              "content": Object {
                "label": "Salvar seu cardápio",
              },
              "type": "next",
            },
          ],
          "subheading": "Aqui está o cardápio que preparei para sua semana",
          "title": "Cardápio semanal",
        },
      }
    }
    resultSize={10}
    stepId="result"
  >
    <div
      className="wizard__title wizard__title"
    >
      <Text
        className=""
        tag={1}
        text="Cardápio semanal"
      >
        <h2
          className=""
        >
          Cardápio semanal
        </h2>
      </Text>
      <Text
        className="subheading"
        tag={7}
        text="Aqui está o cardápio que preparei para sua semana"
      >
        <p
          className="subheading"
        >
          Aqui está o cardápio que preparei para sua semana
        </p>
      </Text>
    </div>
    test
    <div
      className="wizard__buttons"
      key="wizardButtons"
    >
      <Component
        className="wizard__button wizard__button--primary"
        content={
          Object {
            "label": "Salvar seu cardápio",
          }
        }
        key="button-next"
        onClick={[MockFunction]}
      >
        <button
          className="button button wizard__button wizard__button--primary"
          data-componentname="button"
          data-componentview="button"
          disabled={false}
          hidden={false}
          onClick={[Function]}
          type="button"
        >
          <span
            className="button__label"
          >
            Salvar seu cardápio
          </span>
        </button>
      </Component>
    </div>
  </ResultSection>
</div>
`;

exports[`<ResultSection /> matches the snapshot with 0 results 1`] = `
<ResultSection
  actionCallback={[MockFunction]}
  callbacks={
    Object {
      "test": [MockFunction],
    }
  }
  containerClass="wizard--result"
  content={
    Object {
      "noResult": Object {
        "ctas": Array [
          Object {
            "content": Object {
              "label": "Tentar novamente",
            },
            "type": "back",
          },
        ],
        "subheading": "Minha chance de encontrar aumenta quando você não usa muitos filtros de uma vez.
Gostaria de tentar novamente?",
        "title": "=/ Busquei por toda cozinha mas não encontrei o que você procurava!",
      },
      "onResult": Object {
        "ctas": Array [
          Object {
            "content": Object {
              "label": "Salvar seu cardápio",
            },
            "type": "next",
          },
        ],
        "subheading": "Aqui está o cardápio que preparei para sua semana",
        "title": "Cardápio semanal",
      },
    }
  }
  resultSize={0}
  stepId="result"
>
  <div
    className="wizard__title wizard__title"
  >
    <Text
      className=""
      tag={1}
      text="=/ Busquei por toda cozinha mas não encontrei o que você procurava!"
    >
      <h2
        className=""
      >
        =/ Busquei por toda cozinha mas não encontrei o que você procurava!
      </h2>
    </Text>
    <Text
      className="subheading"
      tag={7}
      text="Minha chance de encontrar aumenta quando você não usa muitos filtros de uma vez.
Gostaria de tentar novamente?"
    >
      <p
        className="subheading"
      >
        Minha chance de encontrar aumenta quando você não usa muitos filtros de uma vez.
Gostaria de tentar novamente?
      </p>
    </Text>
  </div>
  <div
    className="wizard__buttons"
    key="wizardButtons"
  >
    <Component
      className="wizard__button wizard__button--primary"
      content={
        Object {
          "label": "Tentar novamente",
        }
      }
      key="button-back"
    >
      <button
        className="button button wizard__button wizard__button--primary"
        data-componentname="button"
        data-componentview="button"
        disabled={false}
        hidden={false}
        onClick={[Function]}
        type="button"
      >
        <span
          className="button__label"
        >
          Tentar novamente
        </span>
      </button>
    </Component>
  </div>
</ResultSection>
`;
